pkg/asset/installconfig/ibmcloud/validation.go

```diff
@@ -13,10 +14,15 @@ import (
 	"github.com/openshift/installer/pkg/types/ibmcloud"
 )
 
+var (
+	ibmCloudClusterNameRegex = regexp.MustCompile(`^([0-9A-Za-z_.-]){1,32}$`)
+)
+
 // Validate executes platform-specific validation.
 func Validate(client API, ic *types.InstallConfig) error {
 	allErrs := field.ErrorList{}
 	platformPath := field.NewPath("platform").Child("ibmcloud")
+	allErrs = append(allErrs, validateClusterName(ic)...)
```

Comment:
This should be done in static validation similar to https://github.com/openshift/installer/blob/794d9514f71f587360b7f755fb1cb6f65f1510c0/pkg/types/validation/installconfig.go#L86-L89.